src/main/java/org/fcrepo/transform/transformations/LDPathTransform.java

```diff
@@ -78,60 +79,38 @@ public LDPathTransform(final InputStream query) {
-     */
-    public static LDPathTransform getNodeTypeTransform(final Node node,
-        final String key) throws RepositoryException {
-
-        final Node programNode = node.getSession().getNode(CONFIGURATION_FOLDER + key);
-
-        LOGGER.debug("Found program node: {}", programNode.getPath());
-
-        final NodeType primaryNodeType = node.getPrimaryNodeType();
-
-        final Set<NodeType> supertypes = orderedBy(BY_NAME).add(primaryNodeType.getSupertypes()).build();
-        final Set<NodeType> mixinTypes = orderedBy(BY_NAME).add(node.getMixinNodeTypes()).build();
-
-        // start with mixins, primary type, and supertypes of primary type
-        final ImmutableList.Builder<NodeType> nodeTypesB = builder();
-        nodeTypesB.addAll(mixinTypes).add(primaryNodeType).addAll(supertypes);
-
-        // add supertypes of mixins
-        mixinTypes.stream().map(mixin -> orderedBy(BY_NAME).add(mixin.getDeclaredSupertypes()).build())
-            .forEach(nodeTypesB::addAll);
-
-        final List<NodeType> nodeTypes = nodeTypesB.build();
-
-        LOGGER.debug("Discovered node types: {}", nodeTypes);
-        for (final NodeType nodeType : nodeTypes) {
-            if (programNode.hasNode(nodeType.toString())) {
-                return new LDPathTransform(programNode.getNode(nodeType.toString())
-                                               .getNode(JCR_CONTENT)
-                                               .getProperty(JCR_DATA)
-                                               .getBinary().getStream());
-            }
-        }
-
-        throw new WebApplicationException(new Exception(
-                "Couldn't find transformation for " + node.getPath()
-                        + " and transformation key " + key), SC_BAD_REQUEST);
+    public static LDPathTransform getResourceTransform(final FedoraResource resource, final NodeService nodeService,
+            final String key) throws RepositoryException {
+
+        final FedoraResource programResource = nodeService.find(resource.getNode().getSession(), CONFIGURATION_FOLDER + key);
+    
+        LOGGER.debug("Found program resource: {}", programResource.getPath());
+    
+        final List<URI> rdfTypes = resource.getTypes();
+    
+        LOGGER.debug("Discovered rdf types: {}", rdfTypes);
+        
+        final List<String> rdfStringTypes = rdfTypes.stream().map(type -> programResource.getPath() + "/" + type.toString().replace( REPOSITORY_NAMESPACE, "fedora:") + "/jcr:content").collect(Collectors.toList());
+        
+        final FedoraBinary transform = (FedoraBinary) programResource.getChildren()
+            .peek(child -> LOGGER.debug("programResource child path is {}", child.getPath()))
+            .filter(child -> rdfStringTypes.contains(child.getPath()))
+            .findFirst()
+            .orElseThrow(() -> new TransformNotFoundException(String.format("Couldn't find transformation for {} and transformation key {}", resource.getPath(), key)));
```

Comment:
Just a side note: this statement shows that you've really grokked the Stream API. @whikloj ++!